fabric-recipe-api-v1/src/client/java/net/fabricmc/fabric/impl/recipe/sync/client/RecipeSyncImplClient.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.recipe.sync.client;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+
+import net.minecraft.recipe.RecipeEntry;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
+import net.fabricmc.fabric.api.client.recipe.v1.sync.ClientRecipeSynchronizedEvent;
+import net.fabricmc.fabric.api.recipe.v1.sync.SynchronizedRecipes;
+import net.fabricmc.fabric.impl.recipe.sync.RecipeSyncPayloadS2C;
+import net.fabricmc.fabric.impl.recipe.sync.SynchronizedRecipesImpl;
+
+public class RecipeSyncImplClient implements ClientModInitializer {
+	@Override
+	public void onInitializeClient() {
+		ClientPlayNetworking.registerGlobalReceiver(RecipeSyncPayloadS2C.ID, RecipeSyncImplClient::onRecipeSyncPacket);
+	}
+
+	private static void onRecipeSyncPacket(RecipeSyncPayloadS2C payload, ClientPlayNetworking.Context context) {
+		SynchronizedRecipes recipes;
+
+		if (!payload.entries().isEmpty()) {
+			var collectedRecipes = new ArrayList<RecipeEntry<?>>();
+
+			for (RecipeSyncPayloadS2C.Entry entry : payload.entries()) {
+				collectedRecipes.addAll(entry.recipes());
+			}
```

Comment:
Are we happy to always sync all recipes? Is there a use case where only a small handful of new recipes may want syncing. For example if you only wanted the client to know about recipes unlocked via advancements. I believe there are also mods that may remove recipes dynamically as well.

Reply:
For recipes unlocked via advancements / that are dynamic it probably makes more sense to use vanilla recipe book system instead. This is useful in cases where you always want all recipes to be synced (think stonecutter, recipe viewers, etc)